buildSrc/src/main/kotlin/IntegrationTestsTasks.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import de.undercouch.gradle.tasks.download.Download
+import org.gradle.api.Project
+import org.gradle.api.file.RelativePath
+import org.gradle.api.tasks.Copy
+import org.gradle.kotlin.dsl.register
+
+class ProjectToTest(val name: String, val hash: String, val organization: String = "Kotlin")
+
+private val projectsToTest = listOf(
+    ProjectToTest("kotlinx.collections.immutable", "592f05fce02a1ad9e26cc6f3fdb55cdd97910599")
+)
+
+fun Project.registerIntegrationTestsTasks() {
```

Comment:
Consider renaming to `registerIntegrationTestsPrerequisites` so that the name matches with the registered gradle task name.